ansible-ipi-install/roles/installer/tasks/24_rhcos_image_cache.yml

```diff
@@ -0,0 +1,87 @@
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    mode: '0755'
+
+# rhcos_json fact already set in 23_rhcos_image_paths.yaml
+- name: Set facts for RHCOS_QEMU_SHA256 and RHCOS_SHA256
+  set_fact:
+    rhcos_qemu_sha256: '{{ rhcos_json.json | json_query(''images.qemu."uncompressed-sha256"'') }}'
+    rhcos_sha256: "{{ rhcos_json.json | json_query('images.openstack.sha256') }}"
+
+- name: Check if {{ rhcos_qemu_uri }} is already in cache
+  stat: 
+    path: "{{ provision_cache_store }}{{ rhcos_qemu_uri }}"
+    checksum_algorithm: sha256
+    get_checksum: yes
+  register: rhcos_qemu_stat
+  when: bootstraposimage is not defined or bootstraposimage|length < 1
+
+- name: Download {{ rhcos_qemu_uri }} for cache
+  get_url:
+    url:  "{{ rhcos_path }}{{ rhcos_qemu_uri }}"
+    dest: "{{ provision_cache_store }}{{ rhcos_qemu_uri }}"
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    mode: '0644'
+    timeout: 3600
+  when: (bootstraposimage is not defined or bootstraposimage|length < 1) and (rhcos_qemu_stat.stat.exists == false or rhcos_qemu_stat.stat.checksum != rhcos_qemu_sha256)
+
+- name: Check if {{ rhcos_uri }} is already in cache
+  stat: 
+    path: "{{ provision_cache_store }}{{ rhcos_uri }}"
+    checksum_algorithm: sha256
+    get_checksum: yes
+  register: rhcos_stat
+  when: clusterosimage is not defined or clusterosimage|length < 1
+
+- name: Download {{ rhcos_uri }} for cache
+  get_url:
+    url:  "{{ rhcos_path }}{{ rhcos_uri }}"
+    dest: "{{ provision_cache_store }}{{ rhcos_uri }}"
+    owner: "{{ ansible_user }}"
+    group: "{{ ansible_user }}"
+    mode: '0644'
+    timeout: 3600
+  when: (clusterosimage is not defined or clusterosimage|length < 1) and (rhcos_stat.stat.exists == false or rhcos_stat.stat.checksum != rhcos_sha256)
+
+- name: Set bootstrap image URL override if not provided by the user
+  set_fact:
+    bootstraposimage: "http://172.22.0.1/{{ rhcos_qemu_uri }}?sha256={{ rhcos_qemu_sha256 }}"
+  when: bootstraposimage is not defined or bootstraposimage|length < 1
+
+- name: Set cluster image URL override if not provided by the user
+  set_fact:
+    clusterosimage: "http://172.22.0.1/{{ rhcos_uri }}?sha256={{ rhcos_sha256 }}"
+  when: clusterosimage is not defined or clusterosimage|length < 1
+
+# install-config.yaml is normally generated via the node-prep role, but we need
+# to regenerate it here to incorporate cache values.  We have to do the cache value
+# generation in the installer role, because this role is what downloads the installer
+# binary.  The installer binary is required to determine the commit ID used for the 
+# selected OCP version, and the commit ID is needed to ultimately determine the image 
+# URIs.
+- name: Re-do install-config.yaml generation with cache values
+  include_role:
+    name: node-prep
+    tasks_from: 100_create-install-config
+
+- name: Start RHCOS image cache container
+  podman_container:
+    name: rhcos_image_cache
+    image: httpd
```

Comment:
I think we better use the RHEL8 httpd image instead the docker.io one. See https://access.redhat.com/containers/#/registry.access.redhat.com/rhel8/httpd-24

Reply:
And if so, the volumes shall be `"{{ provision_cache_store }}:/var/www"`

Reply:
Ack.  Let's use the RHEL 8 image as you suggest.

Reply:
@e-minguez Unfortunately the RHEL 8 httpd image in the redhat.io seems to require authentication.  That would add a barrier to entry to using the cache that we probably don't want.  So for now it looks like we'll stick with the Docker version.

Reply:
Bummer. What about using the CentOS one? https://registry.centos.org/centos/httpd-24-centos7/latest or https://registry.centos.org/centos/httpd/latest

Reply:
Sure, let me give that a try

Reply:
@e-minguez It worked!  Thanks for the suggestion :+1: